File: src/test/java/de/spries/fleetcommander/model/facade/OtherPlayerTest.java
package de.spries.fleetcommander.model.facade;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

import org.junit.Before;
import org.junit.Test;

import de.spries.fleetcommander.model.core.Player;

public class OtherPlayerTest {

	private Player originalPlayer;
	private OtherPlayer viewingPlayer;

	@Before
	public void setUp() {
		originalPlayer = mock(Player.class);
		viewingPlayer = new OtherPlayer(originalPlayer);
	}

	@Test
	public void forwardsCallToGetNameForOtherPlayers() {
		viewingPlayer.getName();
		verify(originalPlayer).getName();
	}

	@Test
	public void forwardsCallToIsActive() {
		viewingPlayer.isActive();
		verify(originalPlayer).isActive();
	}
}

File: src/test/kotlin/de/spries/fleetcommander/model/facade/OtherPlayerTest.kt
package de.spries.fleetcommander.model.facade

import com.nhaarman.mockitokotlin2.mock
import de.spries.fleetcommander.model.core.Player
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito.verify

class OtherPlayerTest {

    private lateinit var originalPlayer: Player
    private lateinit var viewingPlayer: OtherPlayer

    @Before
    fun setUp() {
        originalPlayer = mock()
        viewingPlayer = OtherPlayer(originalPlayer)
    }

    @Test
    fun forwardsCallToGetNameForOtherPlayers() {
        viewingPlayer.getName()
        verify(originalPlayer).name
    }

    @Test
    fun forwardsCallToGetStatus() {
        viewingPlayer.getStatus()
        verify(originalPlayer).status
    }

    @Test
    fun forwardsCallToIsHumanPlayer() {
        viewingPlayer.isHumanPlayer()
        verify(originalPlayer).isHumanPlayer()
    }
}
